src/tasks/includes/lint-reporter.js

```diff
@@ -0,0 +1,52 @@
+const gutil = require('gulp-util');
+
+/** Class representing a custom reporter for @shopify/theme-lint */
+export default class Reporter {
+  constructor() {
+    this.successes = [];
+    this.failures = [];
+  }
+
+  /**
+   * Pushes a valid message onto successes.
+   *
+   * @param {String} message
+   * @param {String} file
+   */
+  success(message, file = null, index = null) {
+    this.successes.push([message, file, index]);
+  }
+
+  /**
+   * Pushes an invalid message onto failures.
+   *
+   * @param {String} message
+   * @param {String} file
+   */
+  failure(message, file = null, index = null) {
+    this.failures.push([message, file, index]);
+  }
+
+  /**
+   * Finalizes string output for translation tests
+   * depending on successes and failures.
+   */
+  finalize() {
```

Comment:
Super nit-picky but I don't love `finalize` as the method name here. I think something like `output` might make sense. 

Another idea... could you maybe call this method `.flush` or something like that but then also reset `this.successes` and `this.failures` back to empty arrays? That way it's like you can empty the output whenever you want and resume it afterwards. Might make for a nicer API and more reusable class.